pkg/core/ca/provided/manager.go

```diff
@@ -30,7 +30,7 @@ type ProvidedCa struct {
 }
 
 type ProvidedCaManager interface {
-	AddSigningCert(ctx context.Context, mesh string, root tls.KeyPair) error
+	AddSigningCert(ctx context.Context, mesh string, root tls.KeyPair) (SigningCert, error)
```

Comment:
If `SigningCert` in return is optional, it might be more clear to express with a pointer type than empty object